src/solver/utils/ortools_utils.cpp

```diff
@@ -212,14 +215,35 @@ static void extract_from_MPSolver(MPSolver* solver,
     }
 }
 
+std::filesystem::path CreateRandomSubDir(const std::filesystem::path& parentDir)
+{
+    char template_array[] = "XXXXXX";
+#ifdef __linux__
+    auto template_dir = parentDir / template_array;
+    char* template_dir_array = template_dir.string().data();
+    if (auto ret = mkdtemp(template_dir_array); ret != nullptr)
+    {
+        return ret;
+    }
+#elif _WIN32
+    if (auto ret = _mktemp_s(template_array); ret == 0)
+    {
+        auto created_dir = parentDir / template_array;
+        std::filesystem::create_directory(created_dir);
+        return created_dir;
+    }
+#endif
+    else
+    {
+        Antares::logs.warning() << "Could not create random subdirectory in " << parentDir.string()
+                                << "\n";
+        return parentDir;
+    }
+}
 std::string generateTempPath(const std::string& filename)
 {
-    namespace fs = std::filesystem;
-    std::ostringstream tmpPath;
-    tmpPath << fs::temp_directory_path().string() << Yuni::IO::SeparatorAsString << filename;
-    return tmpPath.str();
+    return (CreateRandomSubDir(std::filesystem::temp_directory_path()) / filename).string().c_str();
```

Comment:
This changes the behavior. How about keeping the same behavior as previously ?

Reply:
see Pr [reviews](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1791)

Reply:
and we may remove the last 2 commits